File: starter/email_parser.py
# this file will handle the parsing of the email

import email


def parse_email(file_path):
    """
    Parse an email file and print its headers and body.

    Args:
        file_path (str): The path to the email file.

    Returns:
        None
    """
    with open(file_path, 'r') as file:
        msg = email.message_from_file(file)

        # access email headers
        print("From", msg['From'])
        print("To", msg['To'])
        print("Subject", msg['Subject'])

        # access email body
        if msg.is_multipart():
            for part in msg.walk():
                # each part is a either non-multipart, or another multipart message
                # that contains further parts... Message is organized like a tree
                if part.get_content_type() == 'text/plain':
                    print(part.get_payload())
        else:
            print(msg.get_payload())

parse_email('email.eml')

File: starter/email_parser.js
import fs from 'fs';
import { MailParser } from 'mailparser';

function parseEmail(filePath) {
    fs.readFile(filePath, (err, data) => {
        if (err) throw err;

        let mailparser = new MailParser();

        mailparser.on("headers", headers => {
            console.log("From: ", headers.get('from').text);
            console.log("To: ", headers.get('to').text);
            console.log("Subject: ", headers.get('subject'));
        });

        mailparser.on("data", data => {
            if (data.type === 'text') {
                console.log("Body: ", data.text);  // prints the text/plain body of the email
            }
        });

        mailparser.write(data);
        mailparser.end();
    });
}

parseEmail('email.eml');